src/astro/utils/first_table.py

```diff
@@ -0,0 +1,96 @@
+import inspect
```

Comment:
Can we rename the module name to `src/astro/utils/first_table.py` -> `src/astro/utils/table.py`.

That way we have all the utils that are shared between dataframe and transform in a single file. Ideally would love to not have utils 😆 but looks like this needs to be there for now, atleast.